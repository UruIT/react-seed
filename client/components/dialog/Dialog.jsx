import React from 'react';
import style from './Dialog.scss';
import Modal from '../modal/Modal';

export default class Dialog extends React.Component {
	constructor(props) {
		super(props);
		this.state = {
			isModalOpen: false,
			enableCloseOnBackgroundClick: true,
			contentWidth: 350,
			contentHeight: 400,
			limitHeight: true
		};
		this.handleChange = this.handleChange.bind(this);
		this.handleSubmit = this.handleSubmit.bind(this);
		this.openModal = this.openModal.bind(this);
		this.closeModal = this.closeModal.bind(this);
	}

	render() {
		return (
			<div className={style.dialog}>
				<label>
					Close modal on background clicking:
					<input
						type="checkbox"
						name="enableCloseOnBackgroundClick"
						checked={this.state.enableCloseOnBackgroundClick}
						onChange={this.handleChange}
					/>
				</label>
				<label>
					Limit modal height (scrolleable modal):
					<input
						type="checkbox"
						name="limitHeight"
						checked={this.state.limitHeight}
						onChange={this.handleChange}
					/>
				</label>
				<label>
					Conent WIDTH (in px):
					<input
						type="text"
						name="contentWidth"
						placeholder="width (px)"
						value={this.state.contentWidth}
						onChange={this.handleChange}
					/>
				</label>
				<label>
					Conent HEIGHT (in px):
					<input
						type="text"
						name="contentHeight"
						placeholder="height (px)"
						value={this.state.contentHeight}
						onChange={this.handleChange}
					/>
				</label>
				<input type="submit" onClick={this.openModal} value="OPEN DIALOG" />
				<Modal
					open={this.state.isModalOpen}
					limitHeight={this.state.limitHeight}
					onClose={this.closeModal}
					closeClickingOutside={this.state.enableCloseOnBackgroundClick}
				>
					<div
						className={style.content}
						style={{ width: this.state.contentWidth, height: this.state.contentHeight }}
					>
						Size: {this.state.contentWidth} px X {this.state.contentHeight} px
						<form onSubmit={this.handleSubmit}>
							<label>
								{' '}Form label
								<input type="text" value="Form input - fix value" />
							</label>
							<input type="submit" value="Submit form" />
						</form>
						<button onClick={this.closeModal}>Close modal (order by content)</button>
					</div>
				</Modal>
			</div>
		);
	}

	handleChange(e) {
		let value;
		if (e.target.type === 'checkbox') {
			value = e.target.checked;
		} else {
			value = e.target.value;
		}
		this.setState({ [e.target.name]: value });
	}

	handleSubmit(e) {
		e.preventDefault();
	}

	openModal() {
		this.setState({ isModalOpen: true });
	}

	closeModal() {
		this.setState({ isModalOpen: false });
	}
}
