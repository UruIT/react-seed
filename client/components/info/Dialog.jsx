import React from 'react';
import style from './Dialog.scss';
import Modal from '../modal/Modal';

export default class Dialog extends React.Component {
	constructor(props) {
		super(props);
		this.state = {
			isModalOpen: false,
			enableCloseOnBackgroundClick: true,
			contentWidth: 350,
			contentHeight: 400
		};
		this.handleChange = this.handleChange.bind(this);
		this.handleSubmit = this.handleSubmit.bind(this);
	}

	render() {
		return (
			<div className={style.dialog}>
				<label>
					Close modal on background clicking:
					<input
						type="checkbox"
						name="enableCloseOnBackgroundClick"
						checked={this.state.enableCloseOnBackgroundClick}
						onChange={this.handleChange}
					/>
				</label>
				<label>
					Conent WIDTH (in px):
					<input
						type="text"
						name="contentWidth"
						placeholder="width (px)"
						value={this.state.contentWidth}
						onChange={this.handleChange}
					/>
				</label>
				<label>
					Conent HEIGHT (in px):
					<input
						type="text"
						name="contentHeight"
						placeholder="height (px)"
						value={this.state.contentHeight}
						onChange={this.handleChange}
					/>
				</label>
				<input type="submit" onClick={() => this.openModal()} value="OPEN DIALOG" />
				<Modal
					open={this.state.isModalOpen}
					onClose={() => this.closeModal()}
					closeClickingOutside={this.state.enableCloseOnBackgroundClick}
				>
					<div
						className={style.content}
						style={{ width: this.state.contentWidth, height: this.state.contentHeight }}
					>
						Size: {this.state.contentWidth} px X {this.state.contentHeight} px
						<form onSubmit={this.handleSubmit}>
							<label> Form label
								<input type="text" value="Form input - fix value" />
							</label>
							<input type="submit" value="ok" />
						</form>
						<button onClick={() => this.closeModal()}>Close</button>
					</div>
				</Modal>
			</div>
		);
	}

	handleChange(e) {
		let value;
		if (e.target.type === 'checkbox') {
			value = e.target.checked;
		} else {
			value = e.target.value;
		}
		this.setState({ [e.target.name]: value });
	}

	handleSubmit(e) {
		// preventDefault will prevent modal from closing
		e.preventDefault();
	}

	openModal() {
		this.setState({ isModalOpen: true });
	}

	closeModal() {
		this.setState({ isModalOpen: false });
	}
}
