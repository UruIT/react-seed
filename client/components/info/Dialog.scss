.dialog {
	text-align: center;

	input[type="submit"] {
		display: block;
		padding: 5px 20px;
		border-radius: 5px;
		margin: 10px auto;
	}

	button {
		margin-top: 20px;
		padding: 10px 30px;
		border-radius: 5px;
	}

	label {
		display: block;
		font-size: 20px;
		padding: 10px;
		margin: 10px;

		input {
			padding: 10px;
			margin: 0 10px;
			border-radius: 5px;
		}

		input[type="checkbox"] {
			width: 25px;
			height: 25px;
		}
	}

	.content {
		border: solid 1px #0A0817;
		text-align: center;
		background-color: #5882FA;
	}
}
