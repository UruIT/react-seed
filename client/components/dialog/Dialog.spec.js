import React from 'react';
import Dialog from './Dialog';
import renderer from 'react-test-renderer';

test('renders Info', () => {
	const tree = renderer.create(<Dialog />).toJSON();

	expect(tree).toMatchSnapshot();
});
