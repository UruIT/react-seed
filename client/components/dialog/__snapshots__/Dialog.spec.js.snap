// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Info 1`] = `
<div
  className="dialog"
>
  <label>
    Close modal on background clicking:
    <input
      checked={true}
      name="enableCloseOnBackgroundClick"
      onChange={[Function]}
      type="checkbox"
    />
  </label>
  <label>
    Limit modal height (scrolleable modal):
    <input
      checked={true}
      name="limitHeight"
      onChange={[Function]}
      type="checkbox"
    />
  </label>
  <label>
    Conent WIDTH (in px):
    <input
      name="contentWidth"
      onChange={[Function]}
      placeholder="width (px)"
      type="text"
      value={350}
    />
  </label>
  <label>
    Conent HEIGHT (in px):
    <input
      name="contentHeight"
      onChange={[Function]}
      placeholder="height (px)"
      type="text"
      value={400}
    />
  </label>
  <input
    onClick={[Function]}
    type="submit"
    value="OPEN DIALOG"
  />
</div>
`;
